include/slate/internal/MatrixStorage.hh

```diff
@@ -503,6 +517,63 @@ public:
         tiles_.at( ij )->receiveCount()--;
     }
 
+    /// Ensures the tile node exists and increments both the tile life and
+    /// recieve count
+    void tilePrepareToRecieve(ij_tuple ij, int life, Layout layout)
```

Comment:
Should this take a device? If the node doesn't exist, it inserts a Host tile. With GPU-aware MPI, we may never need a Host tile. I'm not sure — I didn't see exactly how this interacts with GPU-aware MPI.

Reply:
All of the places that call this function previously allocated the host node, regardless of GPU-aware MPI.  So, this just keeps the existing logic.  But, it would be good to be smarter when using GPU-aware MPI (and `tileRecv` could be converted to avoid the host->dev copy).

Do you want me to add a cleanup of GPU-aware MPI to this PR?  Or should I just put it on my todo list?

Reply:
Let's put that on the todo list.